spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/config/AdminDefaultEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,93 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.admin.config;
+
+
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.config.YamlPropertiesFactoryBean;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+/**
+ * Contributes the values from {@code admin.yml} if it exists, before any of Spring Boot's normal
+ * configuration contributions apply. This has the effect of supplying overridable defaults
+ * to the various Spring Cloud Dataflow Admin SPI implementations
+ * that in turn override the defaults provided by Spring Boot.
+ *
+ * @author Josh Long
+ */
+public class AdminDefaultEnvironmentPostProcessor implements EnvironmentPostProcessor, Ordered {
+
+	protected Resource resource = new ClassPathResource("/admin.yml");
+
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+
+		Map<String, Object> defaults = new HashMap<>();
+		MutablePropertySources existingPropertySources = environment.getPropertySources();
+
+		this.contributeAdminDefaults(defaults);
+
+		String defaultProperties = "defaultProperties";
+
+		if (!existingPropertySources.contains(defaultProperties) ||
+				existingPropertySources.get(defaultProperties) == null) {
+			existingPropertySources.addLast(new MapPropertySource(defaultProperties, defaults));
+		} else {
+			PropertySource<?> propertySource = existingPropertySources.get(defaultProperties);
+			Map mapOfProperties = Map.class.cast(propertySource.getSource());
+			for (String k : defaults.keySet()) {
+				Set setOfPropertyKeys = mapOfProperties.keySet();
+				if (!setOfPropertyKeys.contains(k)) {
+					mapOfProperties.put(k, defaults.get(k));
+					LogFactory.getLog(getClass()).info(k + '=' + defaults.get(k));
+				}
+			}
+		}
+
+	}
+
+	@Override
+	public int getOrder() {
+		return 0;
+	}
+
+	protected void contributeAdminDefaults(
```

Comment:
because you may want to just contribute _default_ properties using a .property file or just plug in a `Map<K,V>` or something besides my very use-case specific assumption that there's a `admin.yml` to plugin. As we talked about with Dave, this is conceptually hard enough a bit of logic that it's going to become a base support in Boot. I can make this private if u like.